lib/cylc/task_events_mgr.py

```diff
@@ -841,6 +841,8 @@ def _setup_custom_event_handlers(self, itask, event, message):
                     "submit_num": itask.submit_num,
                     "id": quote(itask.identity),
                     "message": quote(message),
+                    "batch_sys_name": quote(itask.summary['batch_sys_name']),
+                    "batch_sys_id": quote(itask.summary['submit_method_id'])
```

Comment:
Please check that both `batch_sys_name` and `submit_method_id` are defined in `self.summary` in `TaskProxy.__init__`. Otherwise, it is worth doing `itask.summary.get` here to avoid `KeyError`.

Reply:
good point

Reply:
I went with get, but probably should define during proxy init (going back to that now)...